scripts/polygon_cleanup_tracks.py

```diff
@@ -0,0 +1,116 @@
+       http://www.apache.org/licenses/LICENSE-2.0
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.​
+
+    This sample allows cleanup of track points from a tracks layer based on a spatial relationship to polygon geometry.
+    Requires being an admin to run this script
+"""
+import argparse
+import logging
+import logging.handlers
+import traceback
+import sys
+from arcgis.gis import GIS
+from arcgis.features import FeatureLayer
+from arcgis import geometry
+
+
+def initialize_logging(log_file=None):
+    """
+    Setup logging
+    :param log_file: (string) The file to log to
+    :return: (Logger) a logging instance
+    """
+    # initialize logging
+    formatter = logging.Formatter(
+        "[%(asctime)s] [%(filename)30s:%(lineno)4s - %(funcName)30s()][%(threadName)5s] [%(name)10.10s] [%(levelname)8s] %(message)s")
+    # Grab the root logger
+    logger = logging.getLogger()
+    # Set the root logger logging level (DEBUG, INFO, WARNING, ERROR, CRITICAL)
+    logger.setLevel(logging.DEBUG)
+    # Create a handler to print to the console
+    sh = logging.StreamHandler(sys.stdout)
+    sh.setFormatter(formatter)
+    sh.setLevel(logging.INFO)
+    # Create a handler to log to the specified file
+    if log_file:
+        rh = logging.handlers.RotatingFileHandler(log_file, mode='a', maxBytes=10485760)
+        rh.setFormatter(formatter)
+        rh.setLevel(logging.DEBUG)
+        logger.addHandler(rh)
+    # Add the handlers to the root logger
+    logger.addHandler(sh)
+    return logger
+
+
+def main(arguments):
+    # initialize logger
+    logger = initialize_logging(arguments.log_file)
+    # Create the GIS
+    logger.info("Authenticating...")
+    # First step is to get authenticate and get a valid token
+    gis = GIS(arguments.org_url,
+              username=arguments.username,
+              password=arguments.password,
+              verify_cert=not arguments.skip_ssl_verification)
+    
+    logger.info("Getting location tracking service")
+    try:
+        tracks_layer = gis.admin.location_tracking.tracks_layer
+    except Exception as e:
+        logger.info(e)
+        logger.info("Getting location tracking service failed - check that you are an admin and that location tracking is enabled for your organization")
+        sys.exit(0)
+    
+    logger.info("Getting polygon layer")
+    try:
+        layer = FeatureLayer(url=args.layer_url, gis=gis)
+        json = layer._lyr_json
+    except Exception as e:
+        logger.info(e)
+        logger.info("Layer could not be found based on given input. Please check your parameters again. Exiting the script")
+        sys.exit(0)
+    
+    features = layer.query(where=args.where, out_sr=4326).features
```

Comment:
you should use a projected coordinate system like 3857 web mercator

Reply:
I thought the LTS was 4326? It appears to be working when I use this, but I get a weird JSON error when I use 3857 (haven't dug too far)